test/plugin/test_in_tail.rb

```diff
@@ -2638,4 +2638,278 @@ def test_lines_collected_with_no_throttling(data)
       end
     end
   end
+
+  # https://github.com/fluent/fluentd/issues/3614
```

Comment:
It prefer #4190 instead of #3614 here to clarify the mechanism of this bug.

Reply:
Thanks! I see!
Fixed in https://github.com/fluent/fluentd/pull/4208#discussion_r1241854041.